l3kernel/l3luatex.dtx

```diff
@@ -554,6 +616,61 @@ end
 %    \end{macrocode}
 % \end{macro}
 %
+% \begin{macro}{ltx.utils.require}
+%   Loads a Lua module. This function loads the module just like the standard
+%   Lua global function |require|, with two key differences:
+%   \begin{enumerate}
+%     \item This function is guaranteed not to throw an error.
+%     \item This function does \emph{not} return the loaded module; instead, it
+%           returns |nil| on success and an error message on failure.
+%   \end{enumerate}
+%    \begin{macrocode}
+local function require(name)
+  if package_loaded[name] then
+    return
+  end
+
+  local failure_details = {}
+  for _, searcher in ipairs(package_searchers) do
+    local loader, data = searcher(name)
+    if type(loader) == 'function' then
+      package_loaded[name] = loader(name, data) or true
+      return
+    elseif type(loader) == 'string' then
+      failure_details[#failure_details + 1] = loader
+    end
+  end
+
+  return table_concat(failure_details, '\n')
+end
+
+ltx.utils.require = require
```

Comment:
I'm not sure if we want to make this public at the moment or if we prefer to keep the option to provide a public `require` with better error management later. If having this available is useful, we could maybe use a name which indicates this? Maybe `try_require`?

Reply:
I've made the function private with the name `try_require`.